sdk/textanalytics/Azure.AI.TextAnalytics/src/ClassifyDocumentOperation.cs

```diff
@@ -0,0 +1,293 @@
+        {
+            async Task<Page<ClassifyDocumentResultCollection>> NextPageFunc(string nextLink, int? pageSizeHint)
+            {
+                var response = await _serviceClient.AnalyzeTextJobStatusNextPageAsync(nextLink, pageSizeHint, _idToIndexMap, cancellationToken).ConfigureAwait(false);
+                return Page.FromValues(new List<ClassifyDocumentResultCollection>() { Transforms.ConvertClassifyDocumentResultCollection(response.Value, _idToIndexMap) }, response.Value.NextLink, response.GetRawResponse());
+            }
+
+            return PageableHelpers.CreateAsyncEnumerable(_ => Task.FromResult(_firstPage), NextPageFunc);
+        }
+
+        async ValueTask<OperationState<AsyncPageable<ClassifyDocumentResultCollection>>> IOperation<AsyncPageable<ClassifyDocumentResultCollection>>.UpdateStateAsync(bool async, CancellationToken cancellationToken)
+        {
+            // BUGBUG
+            Response<AnalyzeTextJobState> response = async
+                ? await _serviceClient.AnalyzeTextJobStatusAsync(_jobId, _showStats, null, null, _idToIndexMap, cancellationToken).ConfigureAwait(false)
+                : _serviceClient.AnalyzeTextJobStatus(_jobId, _showStats, null, null, _idToIndexMap, cancellationToken);
+
+            _createdOn = response.Value.CreatedDateTime;
+            _expiresOn = response.Value.ExpirationDateTime;
+            _lastModified = response.Value.LastUpdatedDateTime;
+            _status = response.Value.Status;
+
+            Response rawResponse = response.GetRawResponse();
+
+            if (response.Value.Status == TextAnalyticsOperationStatus.Succeeded)
+            {
+                string nextLink = response.Value.NextLink;
+                _firstPage = Page.FromValues(new List<ClassifyDocumentResultCollection>() { Transforms.ConvertClassifyDocumentResultCollection(response.Value, _idToIndexMap) }, nextLink, rawResponse);
+
+                return OperationState<AsyncPageable<ClassifyDocumentResultCollection>>.Success(rawResponse, CreateOperationValueAsync(CancellationToken.None));
+            }
+            else if (response.Value.Status == TextAnalyticsOperationStatus.Failed)
+            {
+                RequestFailedException requestFailedException = await ClientCommon
+                    .CreateExceptionForFailedOperationAsync(async, _diagnostics, rawResponse, response.Value.Errors)
+                    .ConfigureAwait(false);
+
+                return OperationState<AsyncPageable<ClassifyDocumentResultCollection>>.Failure(rawResponse, requestFailedException);
+            }
+            else if (response.Value.Status == TextAnalyticsOperationStatus.Cancelled)
+            {
+                return OperationState<AsyncPageable<ClassifyDocumentResultCollection>>.Failure(rawResponse,
+                    new RequestFailedException("The operation was canceled so no value is available."));
+            }
+
```

Comment:
ignore if it's handled somewhere else and I don't see it, but what if partially completed is returned?

https://github.com/Azure/azure-rest-api-specs/blob/main/specification/cognitiveservices/data-plane/Language/stable/2022-05-01/analyzetext.json#L2200

Reply:
IIRC, it was my understanding that it shouldn't happen in practice and is usually a service issue, though maybe we should handle it. That said, I was looking at how you handle it in Python, and - even using the symbols it's defined as e.g. `PARTIALLY_COMPLETED` or used in e.g. `_SUCCEEDED` I don't see it used. What do you do with such a result?

Reply:
I don't see "partiallyCompleted" in Java as well, but do see "partiallySucceeded", but seems Java TA just treats it as another success: https://cs.github.com/Azure/azure-sdk-for-java?q=partiallySucceeded+path%3Asdk%2Ftextanalytics

/cc @mssfang

Reply:
It's rare, but I still see it occur in live tests every now and then. Generally one action fails with an InternalServerError and the LRO is reported as "partiallyCompleted". It's important in Python because if we don't account for it, the LRO will hang forever and never get marked as finished (our azure-core impl of the poller doesn't include "partiallyCompleted" since it's not a standard LRO status per the REST guidelines). 

It's used in _FINISHED in that same file you were looking at - it stops the polling so we can raise an exception. If the service ever decides to return an error target like we discussed before, maybe we'd return partial results, but that's currently not the case.

Reply:
I see - so at the very least I should make sure it's terminal but maybe not much more than that.

Reply:
Yeah I think that's enough

Reply:
Yeah. In TA java, it is tread as the LRO as completed.

Reply:
@mssfang were you going to add "partiallyCompleted" support? From what I see, you only have "partiallySucceeded" but as @kristapratico [pointed out](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/cognitiveservices/data-plane/Language/stable/2022-05-01/analyzetext.json#L2200), the swagger specifies "partiallyCompleted".

Reply:
We had some internal discussion as to whether it should be a successful or failed completion. Because Java and .NET return different result types in separate collection properties, we can actually treat these as success; conversely, JavaScript and Python return all results in the original order, and the service does not provide any sort of reference to know which results we need to offset thus JS and Python treat this condition as failure.

IMO, we should have consistent behavior.

I'm going to add support to terminate `partiallyCompleted` as successful completion for this beta release, but will start an internal thread with the service team to get their thoughts on this.